dot/core/messages_test.go

```diff
@@ -50,28 +52,31 @@ func TestService_ProcessBlockAnnounceMessage(t *testing.T) {
 	err := s.Start()
 	require.Nil(t, err)
 
-	expected := &network.BlockAnnounceMessage{
-		Number:         big.NewInt(1),
-		ParentHash:     s.blockState.BestBlockHash(),
-		StateRoot:      common.Hash{},
-		ExtrinsicsRoot: common.Hash{},
-		Digest:         nil,
-		BestBlock:      true,
-	}
-
 	// simulate block sent from BABE session
-	newBlocks <- types.Block{
+	newBlock := types.Block{
 		Header: &types.Header{
 			Number:     big.NewInt(1),
 			ParentHash: s.blockState.BestBlockHash(),
 		},
 		Body: types.NewBody([]byte{}),
 	}
 
-	time.Sleep(testMessageTimeout)
-	require.NotNil(t, net.Message)
-	require.Equal(t, network.BlockAnnounceMsgType, net.Message.(network.NotificationsMessage).Type())
-	require.Equal(t, expected, net.Message)
+	expected := &network.BlockAnnounceMessage{
+		ParentHash:     newBlock.Header.ParentHash,
+		Number:         newBlock.Header.Number,
+		StateRoot:      newBlock.Header.StateRoot,
+		ExtrinsicsRoot: newBlock.Header.ExtrinsicsRoot,
+		Digest:         newBlock.Header.Digest,
+		BestBlock:      true,
+	}
+
+	//setup the SendMessage function
+	net.On("SendMessage", expected)
```

Comment:
what does `.On()` do when there's no `.Return()`? does it just let the mocker know to expect this argument?

Reply:
Yes! ust let the mocker know the function will be called with this argument